ui/storybook/README.md

```diff
@@ -0,0 +1,122 @@
+- Package-specific stories should live in the `src/stories` directory for that package.
+- Stories that provide project-wide documentation or involve components from multiple packages should live in the storybook package (`ui/storybook/stories`).
+
+### What kind of story format should I use?
+
+#### Components
+
+Stories for components should use [Component Story Format (CSF)](https://storybook.js.org/docs/react/api/csf) and follow the `PascalCaseName.stories.tsx` naming scheme.
+
+See [Writing stories in TypeScript](https://storybook.js.org/blog/writing-stories-in-typescript/) for some additional guidance on typing for stories.
+
+#### Text-only documentation
+
+Stories that are primarily text-based documentation (e.g. the `About` pages for each package) should be written using [MDX](https://storybook.js.org/docs/react/writing-docs/mdx) and follow the `PascalCase.stories.mdx` naming scheme.
+
+These may include importing a few small components like the `LinkTo` component from the [Link addon](https://storybook.js.org/addons/@storybook/addon-links/) to link out to other stories.
+
+### Importing package code
+
+#### TLDR;
+
+Stories should import package code by referencing the package, not by making a relative reference. Under the hood, this will reference source code (not compiled code), so we can benefit from some Storybook features like autogenerating documentation from types.
+
+Import like this in a story:
+
+```ts
+// Import from the package. Storybook's webpack config will alias this to the
+// associated source code.
+import { LineChart } from '@perses-dev/components';
+```
+
+Not like this:
+
+```ts
+// Importing directly from the source file will make it easier to forget to
+// export components for consumption from the main export for the package.
+import { LineChart } from '.';
+```
+
+#### Long version
+
+Stories should import from source code (NOT compiled code) because this enables us to automatically generate documentation tables for component props from types. Stories should import from the top level export for that package to help ensure the component was properly exported for consumption from the package when published. The same versions of code need to be used across the storybook to avoid issues when using `context` and singletons.
+
+The project enforces the above contraints using the following tooling:
+
+- **aliases to source for all @perses-dev packages in storybook's webpack config** - E.g. `@perses-dev/components` will alias to `ui/components/src` instead of pointing to `ui/components/dist`). This makes it easy to import from packages in stories in a consistent way while pointing at the top level `src` export. It also ensures that components referenced indirectly in stories (usually a child of the component used directly in the story) point to source code to avoid accidentally mixing and matching source and compiled code, which can cause issues when using `context` and singletons.
```

Comment:
Somewhat unrelated but this is an approach I was thinking about as well when deciding how to do the codemirror packages for the prom fork. If we did this aliasing in the codemirror extension + lezer grammar, there would be no need to publish a separate package but it could still live in its own package folder. Still thinking about it. But for storybook, I think this makes sense.

Reply:
I _think_ you may also be able to do some magic with private packages in turborepo, but I haven't used it before. Definitely something we can play with a bit when we're ready to do that work.